test/suites/integration/httproute_creation_test.go

```diff
@@ -101,12 +99,12 @@ func verifyResourceCreation(
 	targetGroup *vpclattice.TargetGroupSummary,
 	service *v1.Service,
 ) {
-	time.Sleep(3 * time.Minute) // Allow time for Lattice resources to be created
-
-	vpcLatticeService = testFramework.GetVpcLatticeService(ctx, httpRoute)
-	Expect(*vpcLatticeService.DnsEntry).To(ContainSubstring(latticestore.LatticeServiceName(httpRoute.Name, httpRoute.Namespace)))
+	Eventually(func(g Gomega) {
+		vpcLatticeService = testFramework.GetVpcLatticeService(ctx, httpRoute)
+		g.Expect(*vpcLatticeService.DnsEntry).To(ContainSubstring(latticestore.LatticeServiceName(httpRoute.Name, httpRoute.Namespace)))
+		targetGroup = testFramework.GetTargetGroup(ctx, service)
+		g.Expect(*targetGroup.VpcIdentifier).To(Equal(os.Getenv("CLUSTER_VPC_ID")))
+		g.Expect(*targetGroup.Protocol).To(Equal("HTTP"))
+	})
```

Comment:
why replaced with Eventually?

Reply:
The intention is: instead of wait for 3min then check lattice resource only once, it  sleep for some shorter interval and check lattice resource creation more frequently. The k8s controller follow the design pattern of  "reconciliation" and "eventually state", so "eventually" all lattice resource could be created. Usually the controller don't need 3min (180s) to create all lattice resource, maybe it just need 10sec sleep interval * 10 time retry == 100 sec ==1.6 min in eventually block to verify all lattice resources have been created.

Reply:
don't we need Should(Succeed()) here?

Reply:
Anyways I'm also not a fan of `Eventually(func).Should(Succeed())` approach, I would like to have assertions outside the func whenever possible.

If it is hard to do so, I think this kind of `Eventually()` function should be split. We are testing two separate calls here - getting lattice service and target group. It reduces the need of having complex logic inside Eventually() so that we may be able to pull out some.